src/compiler/checker.ts

```diff
@@ -32180,9 +32184,9 @@ namespace ts {
         function assignParameterType(parameter: Symbol, type?: Type) {
             const links = getSymbolLinks(parameter);
             if (!links.type) {
-                const declaration = parameter.valueDeclaration as ParameterDeclaration;
-                links.type = type || getWidenedTypeForVariableLikeDeclaration(declaration, /*reportErrors*/ true);
-                if (declaration.name.kind !== SyntaxKind.Identifier) {
+                const declaration = parameter.valueDeclaration as ParameterDeclaration | undefined;

```

Comment:
The synthetic `rest` parameter has no declaration, and thus no declaration to `getWidenedTypeForVariableLikeDeclaration` for - if we want the type, we just `getTypeOfSymbol` to trigger promotion of the deferred type instead. (technically we'll be reassigning links.type after initial assignment occurs in `getTypeOfSymbol`, but it'll always be to the exact same type since we're using the return value from `getTypeOfSymbol` so it'll be fine)